keras/applications/nasnet.py

```diff
@@ -188,7 +188,7 @@ def NASNet(
         default_size=default_size,
         min_size=32,
         data_format=backend.image_data_format(),
-        require_flatten=True,
+        require_flatten=include_top,
```

Comment:
Please add/update unit test for this.

Reply:
Added a unit test for this case and verified that test passes locally (and fails without the change).